docs/price-migrations-from-first-principles.md

```diff
@@ -0,0 +1,96 @@
+
+Why the 13th of March and not the 13th of Februrary ? The answer is that there must be at least 30 days between having received the letter and the price rise being effective. Therefore, the payment (as per the invoice that will be generated) on 13th Feb will be using the old price, the earliest you can apply the new price to them is the 13th of March.
+
+Because of this requirement of notifying contracted customers, like Alice School, of any price rise at least 30 days in advance, note that for about 44 days (from the Jan 31st to March 13th), Alice school will effectively be paying less per unit than customers coming in the store.
+
+Lukily for you Alice School can afford the new price and accepts the price rise (without cancelling the contract they have with you). 
+
+### Your business expands
+
+A few years later your business has significantly expanded and you now have hundred of schools that request deliveries of delicacies from you. Some need a daily delivery, some weekly delivery etc... Some schools pay monthly, some schools pay quarterly and some schools pay annually.
+
+Let's actually have a look at 3 of those schools
+
+```
+Alice School   | monthly                                   | 13th
+Bob School     | quarterly (January, April, July, October) | 20th
+Charles School | Annualy (March)                           | 23rd 
+```
+
+Alice school is paying monthly on the 13th, as before. Bob School is paying on quarterly basis, on the indicated months, and does so on the 20th on the months they are paying. Charles School pays annually on the 23rd or March.
+
+Today is March 1st 2027. You decide to price rise your croissants from £1.3 to £1.45. You still have the requirement to let all your customers know 30 days in advance of the coming price rise. So you print as many letters as you have customers and each letter tell them that there's going to be a price rise of croissants from £1.3 per unit to £1.45 per unit (how much each school needs to pay at each billing date depends on the billing frequency, and the delivery schedule and the volume they are ordering from you). The letter also indicates when the price rise will be effective. 
+
+Since we are March 1st 2027, here is the date of effective price rise for the three schools we mentioned:
+
+```
+Alice School   | 13th April 2027
+Bob School     | 20th April 2027
+Charles School | 23rd March 2028
+```
+
+March 1st letter delivery means that you can't price rise Alice School on 13th March, so you will have to wait the next monthly payment. In the case of Bob School the next billing date is April 20th, which is more than 30 days after receiving the letter. In the case of Charles School, 23rd March 2027 is too close because that's less than 30 days after the written notification, so you can only price rise them an entire year later, on 23rd March 2028.
```

Comment:
```suggestion
March 1st letter delivery means that you can't price rise Alice School on 13th March, so you will have to wait for the next monthly payment. In the case of Bob School the next billing date is April 20th, which is more than 30 days after receiving the letter. In the case of Charles School, 23rd March 2027 is too close because that's less than 30 days after the written notification, so you can only price rise them an entire year later, on 23rd March 2028.
```